dpp/src/document/state_transition/documents_batch_transition/validation/state/fetch_documents.rs

```diff
@@ -0,0 +1,90 @@
+use std::collections::hash_map::{Entry, HashMap};
+
+use futures::future::join_all;
+use serde_json::json;
+
+use crate::{
+    document::{document_transition::DocumentTransition, Document},
+    get_from_transition,
+    state_repository::StateRepositoryLike,
+    util::string_encoding::Encoding,
+};
+
+pub struct DocumentsFetcher<SR>
+where
+    SR: StateRepositoryLike,
+{
+    state_repository: SR,
+}
+
+pub fn fetch_documents_factory<SR>(state_repository: SR) -> DocumentsFetcher<SR>
```

Comment:
I'd probably just added this as a `new` to `DocumentsFetcher`

Reply:
Good point! As the existence of `DocumentFetcher` is questionable at all. I'll remove the whole structure and leave the validator as a standalone function.